hmac.go

```diff
@@ -41,19 +41,26 @@ func (h hmacer) verifySignature(signature, payload string) (bool, error) {
 	return hmac.Equal([]byte(expectedSignature), []byte(signature)), nil
 }
 
-func (h hmacer) parseSignature(signatureKeyPair string) (string, error) {
-	if !strings.Contains(signatureKeyPair, "|") {
-		return "", SignatureError{"Signature-key pair does not contain |"}
+func (h hmacer) getMatchingSignature(signaturePairs string) (string, string) {
+	pairs := strings.Split(signaturePairs, "&")
+	for _, pair := range pairs {
+		split := strings.Split(pair, "|")
+		if len(split) == 2 && split[0] == h.publicKey {
+			return split[0], split[1]
+		}
 	}
-	split := strings.Split(signatureKeyPair, "|")
-	if len(split) != 2 {
-		return "", SignatureError{"Signature-key pair contains more than one |"}
+	return "", ""
+}
+
+func (h hmacer) parseSignature(signatureKeyPairs string) (string, error) {
+	if !strings.Contains(signatureKeyPairs, "|") {
+		return "", SignatureError{"Signature-key pair does not contain |"}
 	}
-	publicKey := split[0]
+	publicKey, signature := h.getMatchingSignature(signatureKeyPairs)
 	if publicKey != h.publicKey {
```

Comment:
This line is a little odd to keep. The new `getMatchingSignature` function already matches it to the public key, but then we're double checking the match here again rather than just confirming we got a signature.